tests/test_cli.py

```diff
@@ -0,0 +1,102 @@
+"""Tests for the CLI."""
+import os
+import pathlib
+import sys
+from unittest import mock
+
+import pytest
+
+from open_alchemy import cli
+from open_alchemy import exceptions
+
+
+@pytest.mark.parametrize(
+    "specfile",
+    [
+        pytest.param("spec.json", id="valid json specfile"),
+        pytest.param("spec.yml", id="valid yml specfile"),
+        pytest.param("spec.yaml", id="valid yaml specfile"),
+    ],
+)
+@pytest.mark.cli
+def test_validate_specfile_valid(tmp_path, specfile):
+    """
+    GIVEN a valid specification file
+    WHEN validate_specfile is called
+    THEN nothing happens
+    """
+    spec = tmp_path / specfile
+    spec.write_text("")
+    cli.validate_specfile(spec)
+
+
+@pytest.mark.cli
+def test_validate_specfile_invalid_extension(tmp_path):
+    """
+    GIVEN a invalid specification file extension
+    WHEN validate_specfile is called
+    THEN a ValueError exception is raised
+    """
+    spec = tmp_path / "specfile.txt"
+    spec.write_text("")
+    with pytest.raises(exceptions.BaseError):
+        cli.validate_specfile(spec)
+
+
+@pytest.mark.cli
+def test_validate_specfile_does_not_exist(tmp_path):
+    """
+    GIVEN a specification file which does not exist
+    WHEN validate_specfile is called
+    THEN a ValueError exception is raised
+    """
+    spec = tmp_path / "specfile.txt"
+    with pytest.raises(exceptions.BaseError):
+        cli.validate_specfile(spec)
+
+
+@pytest.mark.cli
+def test_build_application_parser():
```

Comment:
Is it possible/ worth checking whether the parser was constructed with the expected arguments?

Reply:
It is, but then we would just be retesting the stdlib (which is already well tested - https://github.com/python/cpython/blob/3.9/Lib/test/test_argparse.py)

Reply:
The intent wasn't to test argparse, just whether all of the expected `subparsers.add_parser` and `build_parser.add_argument` were called. The reason for it is because what if someone accidentally comments out/ removes one of the lines? Or they refactor and forget that an argument was supposed to be there? It would be good to check that all the expected arguments were added and, if not, that a test or something else fails. The counter argument would be that it should be checked in a pull request or as a part of an integration test.